gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixTask.java

```diff
@@ -213,9 +213,17 @@ private Integer getPartitionForHelixTask(TaskDriver taskDriver) {
 
   @Override
   public void cancel() {
-    log.info("Gobblin helix task cancellation invoked.");
+    log.info("Gobblin helix task cancellation invoked for jobId {}.", jobId);
     if (this.task != null ) {
-      this.task.cancel();
+      try {
+        this.task.cancel();
+        log.info("Gobblin helix task cancellation completed for jobId {}.", jobId);
+      } catch (Throwable t) {
+        log.info("Gobblin helix task cancellation for jobId {} failed with exception.", jobId, t);
+        throw new RuntimeException("Gobblin helix task cancellation for jobId " + jobId + " failed with exception.", t);
```

Comment:
Why not wrap it into a Throwable instead of a RuntimeException?

Reply:
the method signature does not throw any exception. RuntimeException is Unchecked exception and can be thrown without changing the signature. Throwable is not Unchecked exception.

Reply:
Ah I see. Makes sense.

Reply:
Can we just do Throwables.propagate(t), which essentially wraps the throwable into a RuntimeException?